parameter/ElementHandle.cpp

```diff
@@ -41,6 +41,36 @@ using std::string;
 using std::mutex;
 using std::lock_guard;
 
+template <class T>
+struct isVector : std::false_type
+{
+};
+
+template <class T>
+struct isVector<std::vector<T>> : std::true_type
+{
+};
+
+template <class T>
+struct remove_ref_cv
+{
+    using type = typename std::remove_cv<typename std::remove_reference<T>::type>::type;
+};
+
+template <class T>
+static typename std::enable_if<not isVector<typename remove_ref_cv<T>::type>::value,
+                               size_t>::type getUserInputSize(T /*values*/)
+{
+    return 0;
+}
+
+template <class T>
+static typename std::enable_if<isVector<typename remove_ref_cv<T>::type>::value, size_t>::type
+getUserInputSize(T values)
+{
+    return values.size();
+}
+
```

Comment:
Consider using SFINAL to test for size() method existence
``` cpp

/** @return 0 by default, ie for non overloaded types. */
 size_t getUserInputSize(...) { return 0; }

/** @return collection size if it has a size() method. does not participate to overload otherwise. */
template <class T, class..., class = decltype(&T::size)>
size_t getUserInputSize(T collection) { return collection.size(); }
```